packages/xrpl/src/sugar/autofill.ts

```diff
@@ -88,6 +93,32 @@ async function autofill<T extends Transaction>(
   return Promise.all(promises).then(() => tx)
 }
 
+async function setNetworkID(client: Client, tx: Transaction): Promise<void> {
+  if (client.networkID === undefined) {
+    await client.setNetworkIDAndBuildVersion()
+  }
+  if (
+    client.networkID !== undefined &&
+    client.networkID > RESTRICTED_NETWORKS
+  ) {
+    // autofill transaction's networkID if either the network is hooks testnet or build version is >= 1.11.0
+    if (
+      (client.buildVersion &&
+        isEarlierRippledVersion(
```

Comment:
can this function be used anywhere else?

Reply:
Not at the moment, I made a generic version comparison since we might need it in the future for rules that only work from certain version or later similar to this